run-hlt-validation

```diff
@@ -39,11 +39,40 @@ cd HLTrigger/Configuration/test/
 ls
 ./runAll.csh IB | tee -a $WORKSPACE/results/runIB.log
 ./examLogs.csh | tee -a $WORKSPACE/results/examLogs.out
+
+INTEGRTESTS_LOG="${WORKSPACE}/results/hltPhase2UpgradeIntegrationTests.log"
 if which hltPhase2UpgradeIntegrationTests 2>/dev/null ; then
-    hltPhase2UpgradeIntegrationTests --parallelJobs $(nproc) | tee -a  $WORKSPACE/results/hltPhase2UpgradeIntegrationTests.log
+	if hltPhase2UpgradeIntegrationTests --help | grep -w -- "--menu " 2>/dev/null ; then # if the "--menu" option is included in this IB
+
+		TEMPDIR="IntergrTest_temp"
+		TEMPFILE="autoHLT.py"
+		mkdir ${TEMPDIR}
+		cp ${CMSSW_RELEASE_BASE}/src/Configuration/HLT/python/${TEMPFILE} ${TEMPDIR}/${TEMPFILE}
+
+		PYTHON_SCRIPT=$(cat << 'EOF'
+from test import autoHLT
+for k, v in autoHLT.items():
+    if "Run4" in k:
+        print(v)
+EOF
+					 )
+
+		mapfile -t menus < <(python3 -c "$PYTHON_SCRIPT")
+
+		for elem in "${menus[@]}"; do
```

Comment:
may be better to call the python3 command directly here instead of using `mapfile` e.g. I think
```
for elem in $(python3 -c 'from Configuration.HLT.autoHLT import autoHLT;[print(v) for k,v in autoHLT.items() if "Run4" in k]') ; do
```
should work